sevsnpmeasure/guest.py

```diff
@@ -101,14 +109,39 @@ def snp_calc_launch_digest(vcpus: int, vcpu_sig: int, ovmf_file: str,
     snp_update_metadata_pages(gctx, ovmf, sev_hashes, vmm_type)
 
     vmsa = VMSA(SevMode.SEV_SNP, ovmf.sev_es_reset_eip(), vcpu_sig, vmm_type)
-    for vmsa_page in vmsa.pages(vcpus):
+    for i, vmsa_page in enumerate(vmsa.pages(vcpus)):
         gctx.update_vmsa_page(vmsa_page)
+        if dump_vmsa is True:
```

Comment:
I think this is more idiomatic, but if you prefer it the current way I'm OK with it too:
```suggestion
        if dump_vmsa:
```